design/ceph-external-cluster.md

```diff
@@ -0,0 +1,122 @@
+# Rook and External Ceph Clusters
+
+Target version: 1.1
+
+Rook was designed for storage consumption in the same Kubernetes cluster as the clients who are consuming the storage. However, this scenario is not always sufficient.
+
+Another common scenario is when Ceph is running in an “external” cluster from the clients. There are a number of reasons for this scenario:
+- Centralized Ceph management in a single cluster with multiple Kubernetes clusters that need to consume storage.
+- Customers already have a Ceph cluster running not in a K8s environment, likely deployed with Ansible, ceph-deploy, or even manually. They should be able to consume this storage from Kubernetes.
+- Fully independent storage for another level of isolation from their K8s compute nodes. This scenario can technically also be accomplished in a single Kubernetes cluster through labels, taints, and tolerations.
+
+## Terminology
+
+| | |
+|---|---|
+| **Local** Cluster  | The cluster where clients are running that have a need to connect to the Ceph storage. Must be a Kubernetes/OpenShift cluster.  |
+| **External** Cluster  | The cluster where Ceph Mons, Mgr, OSDs, and MDS are running, which might have been deployed with Rook, Ansible, or any other method.  |
+
+## Requirements
+
+Requirements for clients in the local cluster to connect to the external cluster include:
+- At least one mon endpoint where the connection to the cluster can be established
+- Admin keyring for managing the cluster
+- Network connectivity from a local cluster to the mons, mgr, osds, and mds of the external cluster:
+  - mon: for the operator to watch the mons that are in quorum
+  - mon/osd: for client access
+  - mgr: for dashboard access
+  - mds: for shared filesystem access
+
+## Rook Operator
+
+When the Rook operator is started, initially it is not aware of any clusters. When the admin creates the operator, they will want to configure the operator differently depending on if they want to configure a local Rook cluster, or an external cluster.
+
+If external cluster management is required, the differences are:
```

Comment:
What if someone wants to use their Kubernetes nodes for OSDs with an existing external cluster?

Reply:
While it might be technically feasible as long as the networking is configured correctly, the proposal is that this wouldn't be supported. The mons, mgr, and osds should be in the same cluster.